src/postamble.js

```diff
@@ -188,13 +190,19 @@ Module['callMain'] = function callMain(args) {
     var start = Date.now();
 #endif
 
+    var ret = Module[
 #if PROXY_TO_PTHREAD
-    // User requested the PROXY_TO_PTHREAD option, so call a stub main which pthread_create()s a new thread
-    // that will call the user's real main() for the application.
-    var ret = Module['_proxy_main'](argc, argv, 0);
+      // User requested the PROXY_TO_PTHREAD option, so call a stub main which pthread_create()s a new thread
+      // that will call the user's real main() for the application.
+      '_proxy_main'
 #else
-    var ret = Module['_main'](argc, argv, 0);
+      '_main'
 #endif
+    ](
+#if MAIN_READS_PARAMS
```

Comment:
This is tricky to read, because we're parameterizing on two `#if`s and this is better than having 4 slightly-different copies of the same statement.

Would this still work if we had earlier
```js
#else // not MAIN_READS_PARAMS
  var argc = 1;
  var argv = 0;
#endif
```
? Because then we can leave the _proxy_main/_main logic alone.

Reply:
Sounds reasonable, yeah - it will add `0,0` to the code size that way, but that might be slightly better for the VM anyhow (to not have to coerce an undefined). I'll change it.